### PR TITLE
Kconfig subs and controllers for uavcan

### PR DESCRIPTION
**Pull request by henrykotze**
Kconfig for subscribers and controllers for uavcan driver. Implemented to address Issue https://github.com/PX4/PX4-Autopilot/issues/21836

### CODE REVIEW

**Comment by dagar** [comment]
I would leave the NXP board alone until we know more, but otherwise I like this approach.
 - per board we should have complete control of every piece of UAVCAN that's actually included
 - the parts included in the build should be runtime configurable so a system is in complete control of what it accepts

Did you see the build failures?

**Comment by henrykotze** [comment]
@dagar Reverted the NXP board to original, and just conformed to same style in Kconfig. 

Not sure why the Rover SITL test would fail. 

Let me know if there is anything else to do here.

**Comment by dirksavage88** [comment]
> @dagar Reverted the NXP board to original, and just conformed to same style in Kconfig.
> 
> Not sure why the Rover SITL test would fail.
> 
> Let me know if there is anything else to do here.

It looks like for some reason my build system did not pick up param.h...I had to manually add it to sensor_bridge.hpp in order to get a successful compile. 

#include <parameters/param.h>
Kazam_screenshot_00010

**Comment by henrykotze** [comment]
@dagar @bkueng  This seems good to go? 

Let me know if there is anything else to do here.

File: src/drivers/uavcan/uavcan_main.cpp [hunk 1]
```diff
@@ -429,8 +452,14 @@ UavcanNode::start(uavcan::NodeID node_id, uint32_t bitrate)
...
 
 	_instance->ScheduleOnInterval(ScheduleIntervalMs * 1000);
+
+#if defined(CONFIG_UAVCAN_ESC_CONTROLLER)
 	_instance->_mixing_interface_esc.ScheduleNow();
+#endif
+
+#if defined(CONFIG_UAVCAN_SERVO_CONTROLLER)
 	_instance->_mixing_interface_servo.ScheduleNow();
+#endif
```
Comment (bkueng) [inline src/drivers/uavcan/uavcan_main.cpp:462]: I would keep these together under a single option, as they're exposed together in the UI as well.

**Comment by dirksavage88** [comment]
Kazam_screenshot_00013
Kazam_screenshot_00016

%95 is better and probably about as good as it will get unless I disable mavlink on the other telemetry port

**Comment by henrykotze** [comment]
Thanks for the flight testing. :raised_hands: 

The parameter update by the automated update of the imu offsets probably caused the CPU load to spike above 95% (which is done only in a disarmed state), but we are running at the threshold for the cpu load. 

@bkueng Anything from your side?

**Review by bkueng — APPROVED** [review]
Looks good

**Pull request merged**